backend/routers/transcribe.py

```diff
@@ -254,43 +247,47 @@ async def receive_audio(dg_socket1, dg_socket2, soniox_socket, speechmatics_sock
         timer_start = time.time()
 
         # nonlocal audio_buffer
-        # audio_buffer = bytearray()
-
-        # speech_state = SpeechState.no_speech
-        # voice_found, not_voice = 0, 0
-        # path = 'scripts/vad/audio_bytes.txt'
-        # if os.path.exists(path):
-        #     os.remove(path)
-        # audio_file = open(path, "a")
+        audio_buffer = bytearray()
+        speech_state = SpeechState.no_speech
+
         try:
             while websocket_active:
                 raw_data = await websocket.receive_bytes()
                 data = raw_data[:]
-                # audio_buffer.extend(data)
+
                 if codec == 'opus' and sample_rate == 16000:
                     data = decoder.decode(bytes(data), frame_size=160)
 
+                audio_buffer.extend(data)
+                if len(audio_buffer) < window_size_bytes:
+                    continue
+
+                speech_state = is_speech_present(audio_buffer[:window_size_bytes], vad_iterator, window_size_samples)
+
+                # if speech_state == SpeechState.no_speech:
+                #     audio_buffer = audio_buffer[window_size_bytes:]
+                #     continue
+
                 if soniox_socket is not None:
-                    await soniox_socket.send(data)
+                    await soniox_socket.send(audio_buffer)
 
                 if speechmatics_socket1 is not None:
-                    await speechmatics_socket1.send(data)
+                    await speechmatics_socket1.send(audio_buffer)
 
                 if deepgram_socket is not None:
                     elapsed_seconds = time.time() - timer_start
                     if elapsed_seconds > duration or not dg_socket2:
-                        dg_socket1.send(data)
+                        dg_socket1.send(audio_buffer)
                         if dg_socket2:
                             print('Killing socket2')
                             dg_socket2.finish()
                             dg_socket2 = None
                     else:
-                        dg_socket2.send(data)
+                        dg_socket2.send(audio_buffer)
 
                 # stream
-                stream_audio(raw_data)
-
-                # audio_buffer = bytearray()
+                stream_audio(audio_buffer)
```

Comment:
if we don't do the post processing, meaning we don't need to storage the audio frame as well. so, the stream audio func is no neccessary. let me remove the redundant logic ;)